include/support.h

```diff
@@ -238,4 +238,10 @@ constexpr size_t static_if_array_then_zero()
 //
 std::string safe_strerror(int err) noexcept;
 
+// Thread-safe replacement for localtime.
+//
+// Uses exactly the same interface as POSIX localtime_r.
+//
+struct tm *safe_localtime(const time_t *timep, struct tm *result);
```

Comment:
If we're providing the same API for `localtime_r`, can we directly provide it and keep the user-side less "wrapped"? 

```
#if defined(WIN32)
struct tm *localtime_r(const time_t *timep, struct tm *result);
#endif
```

Reply:
Right - I went through several iterations of replacement for this function and decided POSIX API picked the overall best solution, but going one step further and avoiding new function name is even better. I moved the function to `cross.h` and replaced `safe_localtime` with `localtime_r`.